runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsPlugin.java

```diff
@@ -64,4 +74,39 @@ StatusBarsConfig provideConfig(ConfigManager configManager)
 	{
 		return configManager.getConfig(StatusBarsConfig.class);
 	}
+
+	@Subscribe
+	public void onGameTick(GameTick gameTick)
+	{
+		if (config.hideAfterCombatDelay() > 0)
+		{
+			shouldToggleStatusBars();
+		}
+		else
+		{
+			overlayManager.add(overlay);
```

Comment:
Instead of adding and removing the overlay various places, you can keep the overlay enabled the entire time and `return;` from it early when the combat has timed out.